grip_api/src/grip_api/task_editor_widgets/socket.py

```diff
@@ -33,30 +32,55 @@ def __init__(self, state, socket_name, index=0, multi_connections=True, count_on
             @param multi_connections: Boolean stating if the socket accepts multiple connections
             @param count_on_this_side: Total number of sockets on this side of the node
         """
-        # Store the id of the object
-        self.id = id(self)
+        super().__init__(socket_name, index, multi_connections)
         # Store all the information
-        self.state = state
-        self.index = index
-        self.name = socket_name
-        self.is_multi_connected = multi_connections
+        self._state = state
+        # Create and store the graphical socket to be displayed
         self.count_on_this_side = count_on_this_side
-        # Compute the position of the socket wrt the state dimensions
-        self.position = self.get_position()
-        # Create the graphical representation of the socket
         self.graphics_socket = GraphicsSocket(self)
-        # Set its pose in the view
-        self.graphics_socket.setPos(*self.position)
-        # List of all the connectors set to this socket
-        self.connectors = list()
-        # Allows to recognize terminal sockets from others when releasing edges
-        self.is_terminal = False
+        self.update_position(count_on_this_side)
+
+    @property
+    def state(self):
+        return self._state
+
+    @state.setter
+    def state(self, state_object):
+        if hasattr(state_object, "graphics_state"):
+            self._state = state_object
+        else:
+            raise TypeError('state must be object of type State')
+
+    @property
+    def count_on_this_side(self):
+        return self._count_on_this_side
+
+    @count_on_this_side.setter
+    def count_on_this_side(self, value):
+        if isinstance(value, int):
+            self._count_on_this_side = value
+        else:
+            raise TypeError('count_on_this_side must be an int')
+
+    @property
+    def position(self):
+        return self._position
+
+    @position.setter
+    def position(self, position_x_y):
+        if isinstance(position_x_y, list) and len(position_x_y) == 2:
+            if isinstance(position_x_y[0], (int, float)) and isinstance(position_x_y[1], (int, float)):
```

Comment:
Could be simplified!
`if isinstance(position_x_y, list) and all(isinstance(value, (int, float)) for value in position_x_y)`

Reply:
Should I include:
`if isinstance(position_x_y, list) and len(position_x_y) == 2 and all(isinstance(value, (int, float)) for value in position_x_y)`
?

Reply:
Sorry didn't publish this earlier and put this to mitigate with the approval

Reply:
changed [here](https://github.com/shadow-robot/sr_grip/pull/16/commits/e8ac01c3bac38f78ed47b322574ebdca12c6bd2f)
included 2 separate tests